app/src/main/java/com/islaharper/dawnofandroid/presentation/components/Widgets.kt

```diff
@@ -0,0 +1,178 @@
+        shadowElevation = AppBarDefaults.TopAppBarElevation,
+        color = MaterialTheme.colorScheme.primary
+    ) {
+        TextField(
+            modifier = Modifier
+                .fillMaxWidth(),
+            value = text,
+            onValueChange = {
+                onTextChanged(it)
+            },
+            placeholder = {
+                Text(
+                    text = "Search...",
+                    modifier = Modifier.alpha(ContentAlpha.medium),
+                    color = MaterialTheme.colorScheme.onPrimary,
+                    style = MaterialTheme.typography.bodyLarge
+                )
+            },
+            textStyle = TextStyle(
+                color = MaterialTheme.colorScheme.onPrimary,
+                fontSize = MaterialTheme.typography.bodyLarge.fontSize
+            ),
+            singleLine = true,
+            leadingIcon = {
+                IconButton(
+                    modifier = Modifier
+                        .alpha(ContentAlpha.medium),
+                    onClick = { }
+                ) {
+                    Icon(
+                        imageVector = Icons.Default.Search,
+                        contentDescription = stringResource(R.string.search_icon),
+                        tint = MaterialTheme.colorScheme.onPrimary
+
+                    )
+                }
+            },
+            trailingIcon = {
+                IconButton(
+                    onClick = {
+                        if (text.isNotEmpty()) {
+                            onTextChanged("")
+                        } else {
+                            onCloseClicked()
+                        }
+                    }
+                ) {
+                    Icon(
+                        imageVector = Icons.Default.Close,
+                        contentDescription = stringResource(R.string.close_icon),
+                        tint = MaterialTheme.colorScheme.onPrimary
+
+                    )
+                }
+            },
+            keyboardOptions = KeyboardOptions(
+                imeAction = ImeAction.Search
+            ),
+            keyboardActions = KeyboardActions(
+                onSearch = {
+                    onSearchClicked(text)
+                    keyboardController?.hide()
+                }
+            ),
+            colors = TextFieldDefaults.textFieldColors(
+                containerColor = Color.Transparent,
+                cursorColor = MaterialTheme.colorScheme.onPrimary
+            )
+        )
+    }
+}
+
+@Preview(name = "Light", showBackground = true)
+@Composable
+fun ThemeSwitcherPreview() {
+    ThemeSwitcher {
+    }
+}
+
+@Preview(name = "Dark", uiMode = Configuration.UI_MODE_NIGHT_YES)
+@Composable
+fun ThemeSwitcherDarkPreview() {
+    ThemeSwitcher(darkTheme = true) {
+    }
+}
+
+@Preview(name = "Light")
+@Preview(name = "Dark", uiMode = Configuration.UI_MODE_NIGHT_YES)
+@Composable
+fun SearchWidgetPreview() {
```

Comment:
Unfortunately, I can't actually review this preview properly because this branch doesn't build. It's good that you were able to break off a piece of work into a branch with a small diff for easy review, but that work should still compile and build so the reviewer can check it out